src/state/state.js

```diff
@@ -188,14 +188,14 @@
          * @ignore
          */
         function _renderFrame(time) {
-            // update all game objects
-            me.game.update(time);
-            // render all game objects
-            me.game.draw();
             // schedule the next frame update
             if (_animFrameId !== -1) {
                 _animFrameId = window.requestAnimationFrame(_renderFrame);
             }
+            // update all game objects
+            me.game.update(time);
+            // render all game objects
+            me.game.draw();
```

Comment:
You're talking about the polyfill we use. In that case, it's better to run the polyfill version of RAF at the end, where it can get a more accurate delta time that properly handles frame dropping. Here's another example case to consider:

* Call `setTimeout(16ms)`
* Code runs for 24ms
* Timeout callback fires >8ms late

When this happens once, it doesn't matter too much. When it happens hundreds of times in a row, the late callbacks add up, and the game actually runs slower. To fix this, polyfill RAF at the end:

* Record current time in polyfill RAF callback
* Code runs in 24ms
* Call `setTimeout(delta)` where `delta` is computed modulus (1000 / 60);

```javascript
var step = 1000 / 60;
var delta = step - ((currentTime - lastTime) % step);
```